cms/djangoapps/contentstore/views/course.py

```diff
@@ -137,7 +137,14 @@ def reindex_course_and_check_access(course_key, user):
     """
     if not has_course_author_access(user, course_key):
         raise PermissionDenied()
-    return CoursewareSearchIndexer.do_course_reindex(modulestore(), course_key)
+
+    course = modulestore().get_course(course_key)
+    groups_usage_info =  GroupConfiguration.get_content_groups_usage_info(modulestore(), course).items()
+    for name, group in groups_usage_info:
+        for module in group:
+            view, args, kwargs = resolve(module['url'])
+            module['usage_key_string'] = kwargs['usage_key_string']
+    return CoursewareSearchIndexer.do_course_reindex(modulestore(), course_key, groups_usage_info)
```

Comment:
I really view this as part of the operation of reindexing - also, the course is already being loaded fully within that indexing task